src/EditorFeatures/Core/Implementation/DocumentationComments/AbstractDocumentationCommentCommandHandler.cs

```diff
@@ -657,20 +658,39 @@ private void InsertExteriorTriviaIfNeeded(ITextView view, ITextBuffer subjectBuf
 
         private void InsertExteriorTrivia(ITextView view, ITextBuffer subjectBuffer, TextLine currentLine, TextLine previousLine)
         {
-            var useTabs = subjectBuffer.GetOption(FormattingOptions.UseTabs);
-            var tabSize = subjectBuffer.GetOption(FormattingOptions.TabSize);
-
-            var firstNonWhitespaceColumn = previousLine.GetColumnOfFirstNonWhitespaceCharacterOrEndOfLine(tabSize);
-            var indentText = firstNonWhitespaceColumn.CreateIndentationString(useTabs, tabSize) + ExteriorTriviaText + " ";
+            var insertionText = CreateInsertionTextFromPreviousLine(previousLine, subjectBuffer);
 
             var firstNonWhitespaceOffset = currentLine.GetFirstNonWhitespaceOffset();
             var replaceSpan = firstNonWhitespaceOffset != null
                 ? TextSpan.FromBounds(currentLine.Start, currentLine.Start + firstNonWhitespaceOffset.Value)
                 : currentLine.Span;
 
-            subjectBuffer.Replace(replaceSpan.ToSpan(), indentText);
+            subjectBuffer.Replace(replaceSpan.ToSpan(), insertionText);
+
+            view.TryMoveCaretToAndEnsureVisible(subjectBuffer.CurrentSnapshot.GetPoint(replaceSpan.Start + insertionText.Length));
+        }
+
+        private string CreateInsertionTextFromPreviousLine(TextLine previousLine, ITextBuffer subjectBuffer)
+        {
+            var useTabs = subjectBuffer.GetOption(FormattingOptions.UseTabs);
+            var tabSize = subjectBuffer.GetOption(FormattingOptions.TabSize);
+
+            var previousLineText = previousLine.ToString();
+            var firstNonWhitespaceColumn = previousLineText.GetColumnOfFirstNonWhitespaceCharacterOrEndOfLine(tabSize);
+
+            var trimmedPreviousLine = previousLineText.Trim();
+            Debug.Assert(trimmedPreviousLine.StartsWith(ExteriorTriviaText), "Unexpected: previous line does not begin with doc comment exterior trivia.");
+
+            // skip exterior trivia.
+            trimmedPreviousLine = trimmedPreviousLine.Substring(3);
+
+            var firstNonWhitespaceOffsetInPreviousXmlText = trimmedPreviousLine.GetFirstNonWhitespaceOffset();
+
+            var extraIndent = firstNonWhitespaceOffsetInPreviousXmlText != null
+                ? trimmedPreviousLine.Substring(0, firstNonWhitespaceOffsetInPreviousXmlText.Value)
+                : " ";
 
```

Comment:
It seems to me that instead of re-building this line from scratch, you could instead just copy the previous line up to the first comment character.

Up to you though.